app/v1/models/domain/TaxYear.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package api.models.domain
+
+import config.FeatureSwitches
+
+/** Opaque representation of a tax year.
+  *
+  * @param value
+  *   A single-year representation, e.g. "2024" represents the tax year 2023-24.
+  */
+final case class TaxYear private (private val value: String) {
+
+  /** The tax year as a number, e.g. for "2023-24" this will be 2024.
+    */
+  val year: Int = value.toInt
+
+  /** The tax year in MTD (vendor-facing) format, e.g. "2023-24".
+    */
+  val asMtd: String = {
+    val prefix  = value.take(2)
+    val yearTwo = value.drop(2)
+    val yearOne = (yearTwo.toInt - 1).toString
+    prefix + yearOne + "-" + yearTwo
+  }
+
+  /** The tax year in the pre-TYS downstream format, e.g. "2024".
+    */
+  val asDownstream: String = value
+
+  /** The tax year in the Tax Year Specific downstream format, e.g. "23-24".
+    */
+  val asTysDownstream: String = {
+    val year2 = value.toInt - 2000
+    val year1 = year2 - 1
+    s"${year1}-$year2"
+  }
+
+  /** Use this for downstream API endpoints that are known to be TYS.
+    */
+  def useTaxYearSpecificApi(implicit featureSwitches: FeatureSwitches): Boolean = featureSwitches.isTaxYearSpecificApiEnabled && year >= 2024
+
+  override def toString: String = s"TaxYear($value)"
+}
+
+object TaxYear {
```

Comment:
Could also include the helper methods (see [here](https://github.com/hmrc/self-assessment-accounts-api/pull/276/files) in TaxYear). If so, will also need to update TaxYearSpec

Reply:
I'm not sure it's best practice to add functionality if it's not going to be used. Perhaps we should leave the taxYear object as it is and add each helper method as we need them (if we ever need them in this API)?